Yoast/wordpress-seo
# Set the data-title attribute, and make it available for translation

## Context


* Modals should have a title.

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where no title was shown on the modal that you see when you click to speed up your site.

## Relevant technical choices:

* I set the `data-title` attribute, so that `$( this ).data( "title" )` will no longer be `undefined`.
* I also removed the `indexation-list-item-presenter`. It was already removed in #14841, but it seems like it came back into the code through a merge conflict.

## Test instructions

This PR can be tested by following these steps:

* Use the Yoast Test Helper to `Reset Indexables tables & migrations`.
* On the admin dashboard, click on the button that says `Click here to speed up your site now`.
* See that the title is `Yoast indexation status`.

## UI changes

## Documentation

## Quality assurance


Fixes https://github.com/Yoast/bugreports/issues/983
